server/json-rpc/src/lib.rs

```diff
@@ -0,0 +1,98 @@
+extern crate jsonrpc_http_server;
+extern crate serde;
+#[macro_use]
+extern crate serde_derive;
+extern crate graph;
+
+use graph::prelude::{JsonRpcServer as JsonRpcServerTrait, *};
+use graph::serde_json;
+use jsonrpc_http_server::{
+    jsonrpc_core::{self, IoHandler, Params, Value},
+    RestApi, Server, ServerBuilder,
+};
+use std::fmt;
+use std::io;
+use std::net::{Ipv4Addr, SocketAddrV4};
+use std::sync::atomic::AtomicBool;
+use std::sync::atomic::Ordering;
+
+#[derive(Debug, Serialize, Deserialize)]
+struct SubgraphAddParams {
+    name: String,
+    ipfs_hash: String,
+}
+
+impl fmt::Display for SubgraphAddParams {
+    fn fmt(&self, f: &mut fmt::Formatter) -> Result<(), fmt::Error> {
+        write!(f, "{:?}", self)
+    }
+}
+
+pub struct JsonRpcServer {}
+
+impl JsonRpcServerTrait for JsonRpcServer {
+    fn serve(
+        port: u16,
+        provider: Arc<impl SubgraphProvider>,
+        logger: Logger,
+    ) -> Result<Server, io::Error> {
+        let addr = SocketAddrV4::new(Ipv4Addr::new(0, 0, 0, 0), port);
```

Comment:
How so? I have to say I don't understand the difference very well. I replicated what @Jannis did for the GraphQL server.

Reply:
The problem with 127.0.0.1 is that it won't accept connections from the outside world, right? That would be problematic, since we'll want to deploy subgraphs to hosted nodes soon. (Without auth initially)

Reply:
Yup, 127.0.0.1 would accept connections only from localhost (loopback interface only).

I'll trust that you two have thought this through then :)